embedded/remotestorage/s3/s3.go

```diff
@@ -106,19 +119,30 @@ func Open(
 		prefix = prefix + "/"
 	}
 
-	return &Storage{
-		endpoint:    endpoint,
-		accessKeyID: accessKeyID,
-		secretKey:   secretKey,
-		bucket:      bucket,
-		location:    location,
-		prefix:      prefix,
+	s3storage := &Storage{
+		endpoint:      endpoint,
+		S3RoleEnabled: S3RoleEnabled,
+		s3Role:        s3Role,
+		accessKeyID:   accessKeyID,
+		secretKey:     secretKey,
+		bucket:        bucket,
+		location:      location,
+		prefix:        prefix,
 		httpClient: &http.Client{
 			CheckRedirect: func(req *http.Request, via []*http.Request) error {
 				return http.ErrUseLastResponse
 			},
 		},
-	}, nil
+		awsInstanceMetadataURL: "http://169.254.169.254",
```

Comment:
I wasn't clear with my previous comment, urls like this should not be part of the code, if an url different than the s3 host is needed then there should be another flag for that